### PR TITLE
chore(clients/java): enforce API stability in Java code

### LINKED ISSUE #3656: Enforce backwards compatibility in Java client
**Description**

Enforce stability in Java client API
https://www.mojohaus.org/clirr-maven-plugin

### PR DESCRIPTION
**Pull request by miguelpires**
## Description

Checks for backwards compatibility against 0.22.0
## Related issues

closes #3656 

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Comment by MiguelPires** [comment]
[This link](http://clirr.sourceforge.net/clirr-core/exegesis.html) contains a list of the changes that are flagged as errors by Clirr

File: parent/pom.xml [hunk 1]
```diff
@@ -102,9 +102,13 @@
     <plugin.version.dependency>3.1.1</plugin.version.dependency>
     <plugin.version.spotbugs>3.1.12.2</plugin.version.spotbugs>
     <plugin.version.sonar>3.7.0.1746</plugin.version.sonar>
+    <plugin.version.clirr>2.8</plugin.version.clirr>
 
     <!-- maven extensions -->
     <extension.version.os-maven-plugin>1.6.2</extension.version.os-maven-plugin>
+    
+    <!-- version against which backwards compatibility is checked -->
+    <backwards.compat.version>0.22.0</backwards.compat.version>
```
Comment (npepinpe) [inline parent/pom.xml]: Will this be updated when we release, or do we need to update this manually? I assume the latter?

**Review by npepinpe — COMMENTED** [review]
Looks good, just one question about how we pick up the version to compare against.

Also, can we rename the PR so it's more reflective of what it does? It says enforce for the client but really it does for a bunch of modules.

File: parent/pom.xml (hunk 1 above)
Comment (MiguelPires) [reply parent/pom.xml]: Thanks for pointing this out. Please check the latest commit, when releasing the version will be updated to the release version

**Review by npepinpe — DISMISSED** [review]
:+1:

**Review by npepinpe — APPROVED** [review]

**Pull request merged**